src/frontend/store/data-context.js

```diff
@@ -0,0 +1,44 @@
+
+/**
+ * WordPress dependencies
+ */
+import { createContext, useContext, useState, useEffect } from '@wordpress/element';
+import apiFetch from '@wordpress/api-fetch';
+import { format } from '@wordpress/date';
+
+const EventDataContext = createContext( [] );
+
+export function useEventData() {
+	const context = useContext( EventDataContext );
+	if ( context === undefined ) {
+		throw new Error( 'useEventData must be used within a EventDataProvider' );
+	}
+	return context;
+}
+
+export function EventDataProvider( { children, data } ) {
+	const [ eventData, setEventData ] = useState( [] );
+	// we bail out early if data is hydrated from the server.
+	if ( data ) {
+		return (
+			<EventDataContext.Provider value={ data }>
+				{ children }
+			</EventDataContext.Provider>
+		);
+	}
```

Comment:
You're [breaking a rule of hooks here](https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level) by bailing early. Instead you should put your condition inside the effect and bail early from the effect instead.

Reply:
Wouldn’t be right if I was running the hook inside an `else` clause? I do agree that having everything inside the useEffect would result in the same result but I guess I'm missing the hook rule part.

Reply:
Agree with @nerrad, hooks rules and we can dry up the code by not having to repeat the `EventDataContext` twice. As he mentioned, just check for `data` on the first line of the `useEffect` and return if its hydrated.